### ISSUE TITLE
In-band lifetime syntax inconsistency?

### ISSUE THREAD

**Issue by HadrienG2** [rust-lang-nursery/edition-guide#37]
In the [in-band lifetime](https://rust-lang-nursery.github.io/edition-guide/2018/transitioning/ownership-and-lifetimes/in-band-lifetimes.html) section, the following syntax is introduced:

```
fn two_args(foo: &Foo, bar: &'bar Bar) -> &'bar Baz
```

Later, in the [lifetime elision in impl](https://rust-lang-nursery.github.io/edition-guide/2018/transitioning/ownership-and-lifetimes/lifetime-elision-in-impl.html) section, it is said that the feature interacts with in-band lifetime syntax in the following way:

```
fn bar(&self, arg: &str) -> &'arg str
```

There is a little difference between these two syntaxes, however. The former syntax gives the impression that the lifetime parameter is named "bar" by choice, but could be anything, whereas the latter syntax gives the impression that one can use `&'arg` as a shorthand to denote the lifetime of reference argument "arg" of a function.

Which syntax is the correct one?

**Comment by Nemo157** [comment]
The [RFC](https://github.com/rust-lang/rfcs/blob/master/text/2115-argument-lifetimes.md) itself appears to be inconsistent. The summary and guide level explanation use the first form (lifetimes must be introduced as part of the arguments, and don't have to use the same name) while the reference level explanation use the second form (lifetimes with the name of an argument binding implicitly refer to that binding).

[Testing the feature](https://play.rust-lang.org/?gist=1b8754ccb62b7de8c105d70d740a906f&version=nightly&mode=debug) seems to show that only the first form works.

**Comment by HadrienG2** [comment]
I guess this is one thing to be corrected before the feature lands, then :)

**Comment by Nemo157** [comment]
Opened https://github.com/rust-lang/rfcs/pull/2486 to fix the RFC and #38 to fix the docs.
